packages/internal-playwright-helpers/src/fixtures.ts

```diff
@@ -0,0 +1,60 @@
+import { Page, test as base } from "@playwright/test";
+import { getBrowserTestingEnvironment } from "@inrupt/internal-test-env";
+import { TestPage } from "./pages/testApp";
+import { OpenIdPage } from "./pages/open-id";
+import { CognitoPage } from "./pages/cognito";
+
+export type TestOptions = {
+  loginOnStart: boolean;
+};
+
+/**
+ * Start the app, and if applicable initiate login from the client, get redirected
+ * to the OpenID provider, log in, allow the client when redirected to the broker,
+ * and completes login after being redirected to client.
+ */
+const initializeTestApp = async (
+  page: Page,
+  openidProvider: string,
+  loginOnStart: boolean,
+  login: string,
+  password: string
+): Promise<void> => {
+  const testPage = new TestPage(page, openidProvider);
+  // Note: these steps must execute in series, not parallel, which is what
+  // Promise.all would do:
+  await testPage.start()
+  if (loginOnStart) {
+    const cognitoPage = new CognitoPage(page);
+    const openIdPage = new OpenIdPage(page);
+    await testPage.startLogin();
+    await cognitoPage.login(login, password);
+    await openIdPage.allow();
+    await testPage.handleRedirect();
+  }
+};
+
+// TODO: write the loginAndDeny function
+// async loginAndDeny(login: string, password: string): Promise<void> => {};
+
+export const test = base.extend<TestOptions>({
+  loginOnStart: [true, { option: true }],
+  // Override the page fixture to start the app automatically
+  page: async ({ page, loginOnStart }, use) => {
+    const { idp, clientCredentials } = getBrowserTestingEnvironment({
+      clientCredentials: {
+        owner: { login: "", password: "" },
+      },
+    });
+    await initializeTestApp(
+      page,
+      idp,
+      loginOnStart,
+      clientCredentials.owner.login,
+      clientCredentials.owner.password
+    );
+    await use(page);
```

Comment:
I think here we can actually do `await use(testPage)` such that when using this we can just do `page.loginAndAllow()` or `page.loginAndDeny()`

Reply:
That is, we're replacing the `page` fixture with our own augmented page fixture.

Reply:
I'm not sure that will be an option: the `page` fixture signature expects an implementation of `Page` to be used, which `TestPage` isn't, it's a wrapper around it. If the tests require access to test-specific controls, then rather than overriding the fixture we should probably add a different one altogether typed as `TestApp`. However, if we do this, then we probably don't want to override the existing `page` fixture at all, but that means that the app will be started and logged in if and only f that new fixture is imported in the tests, e.g.
```
test("...", async ({page, testApp}) => {
  testApp.loginAndAllow();
});
```
However, that requires updating all the existing tests to include `testApp` in the destructured object, **even if it is'nt used in the test** so that the app automatically starts. That's what bit me when trying this out, and that's why I went towards overriding the `page` fixture, which was less awkward than having the unused object in the test call I think

Reply:
As discussed in a follow-up conversation, we will approach this by having a separate `auth` fixture with high-level auth-related operations (e.g. complete login process), and having the base `page` fixture just automatically start the app (but not log in). This comes from not being able to extend the `Page` class, resulting in the `page` fixture not to expose additional methods.